app/src/main/kotlin/com/wire/android/ui/common/bottomsheet/ModalSheetHeaderItem.kt

```diff
@@ -40,30 +42,33 @@ fun ModalSheetHeaderItem(header: MenuModalSheetHeader = MenuModalSheetHeader.Gon
         MenuModalSheetHeader.Gone -> {
             Spacer(modifier = Modifier.height(dimensions().modalBottomSheetNoHeaderVerticalPadding))
         }
+
         is MenuModalSheetHeader.Visible -> {
-            Row(
-                verticalAlignment = Alignment.CenterVertically,
-                modifier = Modifier.padding(
-                    start = dimensions().modalBottomSheetHeaderHorizontalPadding,
-                    end = dimensions().modalBottomSheetHeaderHorizontalPadding,
-                    top = dimensions().modalBottomSheetHeaderVerticalPadding,
-                    bottom = header.customBottomPadding ?: dimensions().modalBottomSheetHeaderVerticalPadding
-                )
-            ) {
-                header.leadingIcon()
-                Spacer(modifier = Modifier.width(dimensions().spacing8x))
-                Text(
-                    text = header.title,
-                    style = MaterialTheme.wireTypography.title02
-                )
+            CompositionLocalProvider(LocalContentColor provides MaterialTheme.colorScheme.secondary) {
```

Comment:
Not really in the scope of this PR, but this is fixing last remaining issue for dark mode on bottom sheet menus title not being readable enough.

Reply:
I think there's a deeper problem and these text colors are just one of the bad effects of that. The issue is that we still use many material2 components but we only support theme from material3.
In this case, we set our custom `sheetBackgroundColor` and by default it tries to guess the correct `sheetContentColor` by comparing it to material2 color scheme, but we don't specify it, we only have material3 color scheme specified in our theme.
But even adding material2 theme or setting custom `sheetContentColor` won't work because it will set `LocalContentColor` for material2, so if we use `Text` from material3, its content won't use this color.
This can become a problem at any place where we still mix material2 and material3 components, so we should migrate to material3 only.
I can see that we still use these components from material2:
```
SwipeToDismiss
DropdownMenu
ModalBottomSheet
Icon
Divider
Scaffold
BottomSheetScaffold
ModalDrawer
```
Fortunately all of them are now also available on material3, we just need to upgrade it to 1.1 and migrate components.
https://developer.android.com/jetpack/compose/designsystems/material2-material3

Reply:
Thanks for the detailed explanation 🙏  For now, let's just go with this temporal fix. But I would be very glad to see these migation to material 3 components and simplify the way we handle these color fallbacks in a separated ticket/PR.